analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -2,161 +2,408 @@
     }
+}
 
-    // See https://github.com/SonarSource/sonar-dotnet/issues/2234
-    public class FuncAndActionCases
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/5430
+namespace Repro5430
+{
+    public class SomeClass
     {
-        static void Main(string[] args)
+        private int InlineInitializedField11 = new SomeClass("1", "s1").PrivateOverload("s1");                  // Compliant
+        private int InlineInitializedField12 = new SomeClass("1", "s1").PrivateOverload("s1", "s2");            // Noncompliant
+        private int InlineInitializedField13 = new SomeClass("1", "s1").PrivateOverload(null, "s2");            // Noncompliant
+        private int InlineInitializedField14 = new SomeClass("1", "s1").PrivateOverload(null, new[] { "s2" });  // Compliant
+        private int InlineInitializedField15 = new SomeClass("1", "s1").PrivateOverload("42", "s1", "s2");      // Compliant
+
+        private int InlineInitializedField21 = new SomeClass("1", "s1").InternalOverload("s1");                 // Compliant
+        private int InlineInitializedField22 = new SomeClass("1", "s1").InternalOverload("s1", "s2");           // Noncompliant
+        private int InlineInitializedField23 = new SomeClass("1", "s1").InternalOverload(null, "s2");           // Noncompliant
+        private int InlineInitializedField24 = new SomeClass("1", "s1").InternalOverload(null, new[] { "s2" }); // Compliant
+        private int InlineInitializedField25 = new SomeClass("1", "s1").InternalOverload("42", "s1", "s2");     // Compliant
+
+        private SomeClass(object a, string b) { }
+        private SomeClass(string a, string b) { }
+        public SomeClass(string a, params string[] bs) { }
+
+        private int PrivateOverload(object a, string b) => 1492;
+        public int PrivateOverload(string a, params string[] bs) => 1606;
+
+        protected int ProtectedOverload(object a, string b) => 1493;
+        public int ProtectedOverload(string a, params string[] bs) => 1607;
+
+        private protected int PrivateProtectedOverload(object a, string b) => 1494;
+        public int PrivateProtectedOverload(string a, params string[] bs) => 1608;
+
+        protected internal int ProtectedInternalOverload(object a, string b) => 1495;
+        public int ProtectedInternalOverload(string a, params string[] bs) => 1609;
+
+        internal int InternalOverload(object a, string b) => 1496;
+        public int InternalOverload(string a, params string[] bs) => 1610;
+
+        protected virtual int OverriddenAsProtected(object a, string b) => 1497;
+        public int OverriddenAsProtected(string a, params string[] bs) => 1611;
+
+        protected int ShadowedAsPublic(object a, string b) => 1498;
+        public int ShadowedAsPublic(string a, params string[] bs) => 1612;
+
+        protected int ShadowedAsProtectedInternal(object a, string b) => 1499;
+        public int ShadowedAsProtectedInternal(string a, params string[] bs) => 1613;
```

Comment:
These are very nice named members. With a clear intention and explanation why they are there.